cranelift/codegen/src/isa/aarch64/inst/emit.rs

```diff
@@ -2921,9 +2934,14 @@ impl MachInstEmit for Inst {
                 }
             }
             &Inst::Call { ref info } => {
-                if let Some(s) = state.take_stack_map() {
+                let (stack_map, user_stack_map) = state.take_stack_map();
+                if let Some(s) = stack_map {
                     sink.add_stack_map(StackMapExtent::UpcomingBytes(4), s);
                 }
+                if let Some(s) = user_stack_map {
+                    let offset = sink.cur_offset() + 4;
```

Comment:
A little thing but rather than the explicit `+ 4` here, could we put the `push_user_stack_map()` call after the `.put4()` below? That way the order of operations here matches the semantics of "stackmap at end of instruction" more directly.